src/domains/point_set.jl

```diff
@@ -25,34 +27,44 @@ julia> PointSet([(rand(),rand()) for i in 1:100])
 ```
 """
 struct PointSet{T,N} <: SpatialDomain{T,N}
-  coords::Matrix{T}
+  coords::Vector{SVector{N,T}} 
 
   function PointSet{T,N}(coords) where {N,T}
     @assert !isempty(coords) "coordinates must be non-empty"
     new(coords)
   end
 end
 
-PointSet(coords::AbstractMatrix{T}) where {T} =
-  PointSet{T,size(coords,1)}(coords)
+function PointSet(coords::AbstractMatrix{T}) where {T}
+  N = size(coords, 1)
+  points = Vector{SVector{N,T}}()
+  for row in eachcol(coords)
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
+end
```

Comment:
Perhaps this can be simplified to

```julia
PointSet(coords::AbstractMatrix) = PointSet(collect(eachcol(coords)))
```

This constructor could also come after the constructor with tuples.

Reply:
Much better. I had that feeling that I short version would be possible, but not yet proficient with Julia :-)